### PR TITLE
592 improve acceptance test checking of data

### PR DESCRIPTION
**Pull request by ryangrundy7**
# Motivation and Context

We have some flakey acceptance tests that break when relying on the queues being drained. I've updated the tests now to poll the action scheduler database to make sure that all the cases for the action plan are in the database.

# What has changed

- Now poll database to see when the sample is ingested

# How to test?

- Push up a new acceptance tests image to your GCR
- Run acceptance tests in docker dev and on GCP

# Links

[Trello](https://trello.com/c/rXXqezIK)

### CODE REVIEW

**Comment by ryangrundy7** [comment]
[Improve acceptance test checking of data (8)](https://trello.com/c/rXXqezIK/592-improve-acceptance-test-checking-of-data-8)

**Review by AdamHawtin — APPROVED** [review]

**Review by hbrace — APPROVED** [review]
LGTM

File: acceptance_tests/features/receipt.feature [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Feature: Case processor handles receipt message from pubsub service
 
-  Scenario Outline: Reciepted Cases increment ceActualResponses
+  Scenario Outline: Receipted Cases increment ceActualResponses
```
Comment (NickGrantX) [inline acceptance_tests/features/receipt.feature:3]: Good to see spelling mistakes stamped out

File: acceptance_tests/features/steps/action_rules.py [hunk 2]
```diff
@@ -15,15 +16,31 @@
...
+    poll_until_sample_is_ingested_to_action(context)
     # TODO these checks intermittently fail as the queue can occasionally be empty while being drained
     # the sleep is a temporary work around until this is fixed proper
     # (by checking for all the cases in the action scheduler DB)
-    time.sleep(1)
     setup_treatment_code_classified_action_rule(context, action_type)
 
 
+def poll_until_sample_is_ingested_to_action(context):
+    sql_query = """SELECT count(*) FROM actionv2.cases WHERE action_plan_id = %s"""
```
Comment (NickGrantX) [inline acceptance_tests/features/steps/action_rules.py:27]: why is this in triple quotes?

File: acceptance_tests/features/steps/action_rules.py [hunk 3]
```diff
@@ -15,15 +16,31 @@
...
+    conn = psycopg2.connect(f"dbname='{Config.DB_NAME}' user={Config.DB_USERNAME} host='{Config.DB_HOST}' "
+                            f"password={Config.DB_PASSWORD} port='{Config.DB_PORT}'{Config.DB_USESSL}")
+    timeout = time.time() + 60
```
Comment (NickGrantX) [inline acceptance_tests/features/steps/action_rules.py:30]: I'm pretty sure you have to use `timedelta` for this kind of stuff, don't you?

**Review by NickGrantX — APPROVED** [review]
I've eyeballed the code. Need to give it a bit more of a thorough test, but looks good 👍

File: acceptance_tests/features/steps/action_rules.py (hunk 2 above)
Comment (ryangrundy7) [reply acceptance_tests/features/steps/action_rules.py:27]: Was triple quoted in other code when the sql was a bit longer but I've just left it there. Can try without if you want.

File: acceptance_tests/features/steps/action_rules.py (hunk 3 above)
Comment (ryangrundy7) [reply acceptance_tests/features/steps/action_rules.py:30]: I've tested this and it seems to work fine. If you prefer me to use timedelta for it then I can.

File: acceptance_tests/features/steps/action_rules.py (hunk 2 above)
Comment (NickGrantX) [reply acceptance_tests/features/steps/action_rules.py:27]: nah... I've approved it haven't I?

File: acceptance_tests/features/steps/action_rules.py (hunk 3 above)
Comment (NickGrantX) [reply acceptance_tests/features/steps/action_rules.py:30]: well... if you're sure lol

**Pull request merged**